podvm-mkosi/README.md

```diff
@@ -0,0 +1,46 @@
+# Podvm images with mkosi
+
+[mkosi](https://github.com/systemd/mkosi) builds a bootable OS image from scratch. This way, we have full control over every detail of the image, especially over the image format and boot process. On the long run, we will implement fully, bit-by-bit reproducible images with mkosi, and use measured boot and an immutable root FS to ensure the image integrity through remote attestation.
+
+## Prerequisites
+
+Currently, mksoi and other related tools are provided through a [Nix](https://nixos.org/) flake. Nix ensures all tools used in the build of the image are itself reproducible and pinned. mkosi requires a very recent systemd version, so using tools installed on the host is usually not possible.
+
+The recommended way to install Nix on Linux systems is the [Determinate Systems nix-installer](https://github.com/DeterminateSystems/nix-installer).
+
+
+## Building an image
+
+### Building PodVM binaries and place them in the binaries-tree
+
+```sh
+docker buildx build \
+    -t podvm-builder-fedora \
+    --load \
+    - < ../podvm/Dockerfile.podvm_builder.fedora
+
+docker buildx build \
+    --build-arg BUILDER_IMG=podvm-builder-fedora \
+    -o type=local,dest="./resources/binaries-tree" \
+    - < ../podvm/Dockerfile.podvm_binaries.fedora
+```
+
+### Activating the Nix development shell
+
+This puts the dependencies from the Nix flake into your path.
+
+```sh
+nix develop ./..#podvm-mkosi
```

Comment:
On my nix installation I had to use:

```bash
nix develop ./..#podvm-mkosi --extra-experimental-features nix-command --extra-experimental-features flakes
```

also I had to install `zstd` & `uidmap` on my buíld system. Are we able to put this into the nix recipe?

Reply:
Mkosi requires `newuidmap` for rootless user namespaces. This cannot be part of the Nix shell, as you cannot give setuid privileges to something in `/nix/store`.

Reply:
added `zstd` to the shell.